File: frontend/lib/models/Bid.dart
class Bid {
  final String tenderId;
  final String? contractorId;
  final double bidAmount;
  final int completionTimeExcepted;
  final int technicalMatchedCount;
  final double? finalBidScore;

  Bid({
    required this.tenderId,
    this.contractorId,
    required this.bidAmount,
    required this.completionTimeExcepted,
    required this.technicalMatchedCount,
    this.finalBidScore,
  });
  factory Bid.fromJson(Map<String, dynamic> json) {
    return Bid(
      tenderId: json['tender_id'],
      bidAmount: json['bid_amount'],
      completionTimeExcepted: json['completion_time'],
      technicalMatchedCount: json['technical_matched_count'],
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'contractor_id': contractorId,
      'tender_id': tenderId,
      'bid_amount': bidAmount,
      'completion_time': completionTimeExcepted,
      'technical_matched_count': technicalMatchedCount,
    };
  }
}

File: app/Models/Bid.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Bid extends Model
{
    use HasFactory;

    protected $fillable = [
        'tender_id',
        'contractor_id',
        'bid_amount',
        'completion_time',
        'technical_proposal_pdf',
        'technical_matched_count',
        'final_bid_score',
        
    ];

    // العلاقة بين العرض والمناقصة
    public function tender()
    {
        return $this->belongsTo(Tender::class);
    }

    // العلاقة بين العرض والمقاول
  public function contractor()
{
    return $this->belongsTo(Contractor::class, 'contractor_id');
}


}
